judge/models/contest.py

```diff
@@ -77,6 +77,10 @@ class Contest(models.Model):
     view_contest_scoreboard = models.ManyToManyField(Profile, verbose_name=_('view contest scoreboard'), blank=True,
                                                      related_name='view_contest_scoreboard',
                                                      help_text=_('These users will be able to view the scoreboard.'))
+    partially_hide_scoreboard = models.BooleanField(verbose_name=_('Partially hide scoreboard'),
```

Comment:
I think instead of having a new boolean this, we should change the existing `hide_scoreboard` field into a `scoreboard_visibility = (Visible | Hidden_for_duration_of_contest | Hidden_for_duration_of_participation | Hidden) ChoiceField`. Then the migration would map `hide_scoreboard = True` to `scoreboard_visibility = Hidden_for_duration_of_contest`.